parsons/action_kit/action_kit.py

```diff
@@ -463,3 +465,83 @@ def create_generic_action(self, page, email=None, ak_id=None, **kwargs):
                                page=page,
                                return_full_json=True,
                                **kwargs)
+
+    def bulk_upload_csv(self, import_page=None, csv_file=None, autocreate_user_fields=0):
+        """
+        Bulk upload a csv file of new users or user updates.
+        If you are uploading a table object, use bulk_upload_table instead.
+        See `ActionKit User Upload Documentation
+             <https://roboticdogs.actionkit.com/docs/manual/api/rest/uploads.html>`
+        Be careful that blank values in columns will overwrite existing data.
+
+        If you get a 500 error, try sending a much smaller file (say, one row),
+        which is more likely to return the proper 400 with a useful error message
+
+        `Args:`
+            import_page: str
+                The page to post the action. The page short name.
+            csv_file: str or buffer
+                The csv file path or a file buffer object
+                A user_id or email column is required.
+            autocreate_user_fields: bool
+                When True columns starting with "user_" will be uploaded as user fields.
+                See the `autocreate_user_fields documentation
+                  <https://roboticdogs.actionkit.com/docs/manual/api/rest/uploads.html#create-a-multipart-post-request>`
+        `Returns`:
+            dict
+                success: whether upload was successful
+                progress_url: an API URL to get progress on upload processing
+                res: requests http response object
+        """
+        import_page = check_env.check('ACTION_KIT_IMPORTPAGE', import_page)
+        if csv_file is None:
+            csv_file = check_env.check('ACTION_KIT_CSVFILE', csv_file)
+
+        # self.conn defaults to JSON, but this has to be form/multi-part....
+        upload_client = self._conn({'accepts': 'application/json'})
+        if isinstance(csv_file, str):
+            csv_file = open(csv_file, 'r')
+
+        res = upload_client.post(
+            self._base_endpoint('upload'),
+            files={'upload': csv_file},
+            data={'page': import_page,
+                  'autocreate_user_fields': int(autocreate_user_fields)})
+        rv = {'res': res,
+              'success': res.status_code == 201,
+              'progress_url': res.headers.get('Location')}
+        return rv
+
+    def bulk_upload_table(self, import_page, table, autocreate_user_fields=0):
```

Comment:
An (undocumented) norm is that we typically include `table` as the first argument in a method that takes a Parsons table.